cockatrice/src/abstractcarddragitem.cpp

```diff
@@ -22,10 +22,12 @@ AbstractCardDragItem::AbstractCardDragItem(AbstractCardItem *_item,
     } else {
         if ((hotSpot.x() < 0) || (hotSpot.y() < 0)) {
             qDebug() << "CardDragItem: coordinate overflow: x =" << hotSpot.x() << ", y =" << hotSpot.y();
-            hotSpot = QPointF();
-        } else if ((hotSpot.x() > CARD_WIDTH) || (hotSpot.y() > CARD_HEIGHT)) {
+            hotSpot = QPointF{qMax(hotSpot.x(), 0.0), qMax(hotSpot.y(), 0.0)};
```

Comment:
```suggestion
            // Range is x=[0, CARD_WIDTH] and y=[0, CARD_HEIGHT]
            hotSpot = QPointF{qMin(qMax(hotSpot.x(), 0.0), static_cast<qreal>(CARD_WIDTH)),
                          qMin(qMax(hotSpot.y(), 0.0), static_cast<qreal>(CARD_HEIGHT))};
```

Reply:
This case (one coordinate too low and the other too high) is already handled by making the `hotSpot.{x, y}() > CARD_{WIDTH,HEIGHT}` condition a second `if` rather than an `else if`.

If we don't care about the debug messages, we could also just have the following without the `if`s (is this is what you were getting at?):

```c++
// Range is x=[0, CARD_WIDTH] and y=[0, CARD_HEIGHT]
hotSpot = QPointF{qBound(0.0, hotSpot.x(), static_cast<qreal>(CARD_WIDTH)),
                  qBound(0.0, hotSpot.y(), static_cast<qreal>(CARD_HEIGHT))};
```

Actually it should probably be `CARD_WIDTH - 1` and `CARD_HEIGHT - 1` now that I think about it -- from `0` to `CARD_WIDTH` inclusive has `CARD_WIDTH + 1` elements.